lms/services/grouping.py

```diff
@@ -97,6 +109,59 @@ def upsert_canvas_group(  # pylint: disable=too-many-arguments
             )
         )
 
+    def upsert_grouping_memberships(self, user: User, groups: List[Grouping]):
+        """
+        Upserts group memberships.
+
+        :param user:  User the that belongs to the groups
+        :param groups: List of groups the `user` belongs to
+        """
+        for group in groups:
+            if membership := (
+                self._db.query(GroupingMembership)
+                .filter_by(grouping_id=group.id, user_id=user.id)
+                .one_or_none()
+            ):
+                membership.updated = func.now()
+                continue
+
+            group.memberships.append(GroupingMembership(grouping=group, user=user))
+
+    def get_course_groupings_for_user(
+        self,
+        course: Course,
+        user_id: str,
```

Comment:
user_id: str (and not User) as this will be used when grading when we only have the ID. Still possible to do user.user_id when calling if you do have a `User` object.